drivers/sensor/adxl345/adxl345.c

```diff
@@ -0,0 +1,226 @@
+	int rc;
+	u8_t write_buf[2];
+
+	write_buf[0] = reg;
+	write_buf[1] = val;
+	rc = i2c_write(data->i2c_master,
+		       write_buf, 2,
+		       data->i2c_addr);
+	if (rc < 0) {
+		LOG_ERR("Writing register failed\n");
+		return rc;
+	}
+
+	return 0;
+}
+
+static int adxl345_read_sample(struct device *dev,
+			       struct adxl345_sample *sample)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+	s16_t raw_x, raw_y, raw_z;
+	u8_t axis_data[6];
+
+	int rc = i2c_burst_read(data->i2c_master,
+				data->i2c_addr,
+				ADXL345_X_AXIS_DATA_0_REG,
+				axis_data,
+				6);
+
+	if (rc < 0) {
+		LOG_ERR("Samples read failed with rc=%d\n", rc);
+		return rc;
+	}
+
+	raw_x = axis_data[0] | (axis_data[1] << 8);
+	raw_y = axis_data[2] | (axis_data[3] << 8);
+	raw_z = axis_data[4] | (axis_data[5] << 8);
+
+	if (raw_x & (1 << 9)) {
+		raw_x |= ADXL345_COMPLEMENT;
+	}
+	if (raw_y & (1 << 9)) {
+		raw_y |= ADXL345_COMPLEMENT;
+	}
+	if (raw_z & (1 << 9)) {
+		raw_z |= ADXL345_COMPLEMENT;
+	}
+
+	sample->x = (raw_x * 1000) / 32;
+	sample->y = (raw_y * 1000) / 32;
+	sample->z = (raw_z * 1000) / 32;
+
+	return 0;
+}
+
+static void adxl345_accel_convert(struct sensor_value *val, unsigned int sample)
+{
+	val->val1 = sample;
+	val->val2 = 0;
+}
+
+static int adxl345_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+	struct adxl345_sample sample;
+	int samples_count;
+	int rc;
+
+	data->sample_number = 0;
+	rc = adxl345_read_reg(dev, ADXL345_FIFO_STATUS_REG);
+	if (rc < 0) {
+		LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
+		return rc;
+	}
+	samples_count = rc;
+	for (int s = 0; s < samples_count; s++) {
+		rc = adxl345_read_sample(dev, &sample);
+		if (rc < 0) {
+			LOG_ERR("Failed to fetch sample rc=%d\n", rc);
+			return rc;
+		}
+		data->bufx[s] = sample.x;
+		data->bufy[s] = sample.y;
+		data->bufz[s] = sample.z;
+	}
+
+	return samples_count;
+}
+
+static int adxl345_channel_get(struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+
+	if (data->sample_number > 32) {
+		data->sample_number = 0;
+	}
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		adxl345_accel_convert(val, data->bufx[data->sample_number]);
+		data->sample_number++;
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		adxl345_accel_convert(val, data->bufy[data->sample_number]);
+		data->sample_number++;
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		adxl345_accel_convert(val, data->bufz[data->sample_number]);
+		data->sample_number++;
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+		adxl345_accel_convert(val++, data->bufx[data->sample_number]);
+		adxl345_accel_convert(val++, data->bufy[data->sample_number]);
+		adxl345_accel_convert(val,   data->bufz[data->sample_number]);
+		data->sample_number++;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api adxl345_api_funcs = {
+	.attr_set = NULL,
```

Comment:
No need to do this explicitly, `adxl345_api_funcs` is static so it will be initialized to null.